src/xenia/memory.cc

```diff
@@ -353,6 +353,25 @@ BaseHeap* Memory::LookupHeapByType(bool physical, uint32_t page_size) {
 
 VirtualHeap* Memory::GetPhysicalHeap() { return &heaps_.physical; }
 
+void Memory::GetHeapsPageStatsSummary(std::vector<BaseHeap*> provided_heaps,
+                                      uint32_t& unreserved_pages,
+                                      uint32_t& reserved_pages,
+                                      uint32_t& used_pages,
+                                      uint32_t& reserved_bytes) {
+  auto lock = global_critical_region_.Acquire();
+  for (const BaseHeap* heap : provided_heaps) {
+    uint32_t heap_unreserved_pages = heap->ureserved_page_count();
+    uint32_t heap_reserved_pages = heap->reserved_page_count();
+
+    unreserved_pages += heap_unreserved_pages;
+    reserved_pages += heap_reserved_pages;
+    used_pages += ((heap->total_page_count() - heap_unreserved_pages) *
+                   heap->page_size()) /
+                  4096;
```

Comment:
Unless there's reverse engineering data proving otherwise for MmQueryStatistics (that it needs to report the physical memory 4 KB portion count — but then I wouldn't call them "pages"), the raw physical memory doesn't have a concept of paging at all, while virtual memory (0xA… are virtual too, but mapped directly 1:1 to physical) has pages with different sizes (4 KB, 64 KB, 16 MB), therefore you can't call an arbitrary 4 KB region a "page".